nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/services/workmanager/WorkManagerRunWorkInFailure.java

```diff
@@ -0,0 +1,159 @@
+public class WorkManagerRunWorkInFailure {
+    private static final Logger log = LogManager.getLogger(WorkManagerRunWorkInFailure.class);
+
+    public static final String ID = "WorkManager.RunWorkInFailure";
+
+    protected static final long DEFAULT_TIMEOUT_SECOND = 120L;
+
+    protected static final long ASSIGNMENT_TIMEOUT_SECOND = 60L;
+
+    protected volatile long countTotal;
+
+    protected volatile long countSuccess;
+
+    @Param(name = "timeoutSecond", required = false)
+    protected long timeout = DEFAULT_TIMEOUT_SECOND;
+
+    @OperationMethod
+    public Blob run() throws IOException, InterruptedException, TimeoutException {
+        StreamManager streamManager = Framework.getService(StreamService.class).getStreamManager(DEFAULT_LOG_MANAGER);
+        Settings settings = new Settings(1, 1, WorkManagerImpl.DEAD_LETTER_QUEUE_CODEC, getComputationPolicy());
+        StreamProcessor processor = streamManager.registerAndCreateProcessor("RunWorkInFailure", getTopology(),
+                settings);
+        try {
+            countTotal = 0;
+            countSuccess = 0;
+            processor.start();
+            processor.waitForAssignments(Duration.ofSeconds(ASSIGNMENT_TIMEOUT_SECOND));
+            if (!processor.drainAndStop(getTimeout())) {
+                throw new TimeoutException();
+            }
+        } finally {
+            processor.shutdown();
+        }
+        return buildResult();
+    }
+
+    private Blob buildResult() throws IOException {
+        Map<String, Object> result = new HashMap<>();
+        result.put("total", countTotal);
+        result.put("success", countSuccess);
+        return Blobs.createJSONBlobFromValue(Collections.singletonList(result));
+    }
+
+    protected Duration getTimeout() {
+        return Duration.ofSeconds(timeout);
+    }
+
+    protected ComputationPolicy getComputationPolicy() {
+        return new ComputationPolicyBuilder().retryPolicy(new RetryPolicy(ComputationPolicy.NO_RETRY))
+                                             .continueOnFailure(true)
+                                             .build();
+    }
+
+    protected Topology getTopology() {
+        return Topology.builder()
+                       .addComputation(WorkFailureComputation::new,
+                               Collections.singletonList(INPUT_1 + ":" + DEAD_LETTER_QUEUE))
+                       .build();
+    }
+
+    protected class WorkFailureComputation extends AbstractComputation {
+
+        private static final String NAME = "WorkFailure";
+
+        public WorkFailureComputation() {
+            super(NAME, 1, 0);
+        }
+
+        @Override
+        public void processRecord(ComputationContext context, String inputStreamName, Record record) {
+            context.askForCheckpoint();
+            Work work = WorkComputation.deserialize(record.getData());
+            log.warn("Trying to run Work from DLQ: " + work.getCategory() + ":" + work.getId());
```

Comment:
Why WARN? INFO would be enough I think.

Reply:
This is a recovery procedure my guess is that ops want to see some output and they are not going to change the log4j for this, wdyt?

Reply:
In practice our ops team gather even INFO logs. This is best practice to not lose data (the default log4j2.xml is not the typical deployment config).